computeGradientFV3.py

```diff
@@ -191,20 +191,30 @@ def optimizedComputeGradientFV3_Private(self, varField, areaInvD, sid, fluxInteg
 
       def computeGradientFV3(self, varField):
 
+            runInParallel = False
             ## If we have not actually precomputed necessary datastructures already, let us do it now
-            if not self.cachedData:
-                  self.precomputeGradientFV3Data()
+            if runInParallel:
+                  if not self.cachedData:
+                        self.precomputeGradientFV3Data()
+            else:
+                  assert(self.cachedData == True)
 
             SF = np.float64
 
             # Gradients are 3 component vectors
             nc = 3
 
-            computepool = multiprocessing.Pool(processes=self.NPROCS)
-            results = computepool.starmap(self.optimizedComputeGradientFV3_Private, 
-                                           zip(repeat(varField), self.areaInvD, self.sid, self.fluxIntegral))
-            computepool.close()
-            computepool.join()
-            varGradient = np.reshape(np.array(results, dtype=SF).T, (nc, varField.shape[0]))
+            NC = len(self.areaInvD)
+            if runInParallel:
+                  computepool = multiprocessing.Pool(processes=1)#self.NPROCS)
```

Comment:
Is this only using 1 process to do all cacheing?

Reply:
The caching is so that we don't repeat the entire process of looking at the adjacency list for each element and figuring out the right weights for gradient assembly. This was originally done on the fly and so now the required data is cached during `precomputeGradientFV3Data`, which is still run in parallel. This evaluation can be made parallel above but I noticed that the savings are much lighter (the speedup is only nelems/procs).

The current workflow computes the caching of gradient data (`precomputeGradientFV3Data`) in parallel, and then utilizes a parallel loop in CANGAMetricsDriver over the remap iterations. Since metrics for different remap iterations are independent, this is a much higher cost savings IMO. And I couldn't figure out how to do nested parallelism. So we can only have one or the other. Open to suggestions.